### PR TITLE
test(eda): add vitest coverage for access/, rulebook-activations/ & projects/ (AAP-70845)

### PR DESCRIPTION
**Pull request by acruzgon**
## Summary

Add 25 new Vitest test files and enhance 5 existing ones across three EDA directories (`access/`, `rulebook-activations/`, `projects/`) to push code coverage toward the 90% target required by AAP-70845.

All new tests follow the AAA pattern, use MSW for HTTP-level API mocking, `userEvent.setup()` for interactions, and accessible queries (`getByRole` > `getByLabelText` > `getByText`).

### New/enhanced test coverage:

**access/** (8 new, 4 enhanced — ~75 tests)
- EditUser (CreateUser, EditUser, EditCurrentUser forms, password validation)
- CredentialTypeForm (create/edit, permissions warning, DataEditor mock)
- CredentialFormTypes (isFieldRequired, field rendering, hidden fields)
- UserPage, OrganizationPage, CredentialTypePage (page rendering, tabs, loading)
- EdaRolePage, OrganizationForm, TeamForm (expanded coverage)
- useCredentialsTestModal, EdaSelectRolesStep, EdaAddRoles

**rulebook-activations/** (12 new — ~63 tests)
- SourceEventMapFields, ActivationInstancePage, ActivationsToolbar
- RulebookActivationHistory, ManageUsers, UserAccess, TeamAccess
- Hooks: useActivationHistoryActions, useActivationHistoryColumns, useActivationHistoryFilters, useCopyRulebookactivation, useQueryProjects

**projects/** (5 new, 1 expanded — ~31 tests)
- EdaProjectManageUsers, EdaProjectCell, useDeleteProjects
- ProjectUserAccess, ProjectTeamAccess
- ProjectDetails (expanded from 5 to 15 tests)

## Type of Change

- [x] Tests

## Risk Analysis - REQUIRED

- [x] **Low** — Narrowly scoped (e.g., single page fix, styling tweak, documentation, test-only changes, copy/string updates). Minimal risk of unintended side effects.

## Testing

- `tsc --noEmit` passes (only pre-existing chatbot error)
- All new tests pass via `npx vitest run` in `frontend/eda/`
- No new test failures introduced (baseline devel has 45 failures, this branch has 42)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by keithjgrant — DISMISSED** [review]

**Comment by vidyanambiar** [comment]
## Code Review: test(eda) coverage for access/, rulebook-activations/ & projects/

  Verdict: NEEDS_CHANGES
  Score: 9.3/10 (Functionality 10, Security 10, Quality 8)

  Findings

  Major

  1. fireEvent usage — useDeleteCredentialTypes.test.tsx and useDeleteEdaRoles.test.tsx use fireEvent.click() instead of
  userEvent.setup(). Project testing guidelines explicitly require userEvent. Both files also use setTimeout(resolve, 1500)
  for async waits instead of waitFor.

  Minor

  2. No-value tests — 4 files contain tests with no behavioral coverage:
    - EdaAddRoles.test.tsx / EdaSelectRolesStep.test.tsx: only check expect(Component).toBeDefined()
    - EdaTeamRoles.test.tsx: 3 tests only assert expect(container).toBeDefined()
    - useDeleteUser.test.tsx: 3 of 4 tests only assert expect(() => result.current(users)).not.toThrow()
    - useDeleteUser.test.tsx: 3 of 4 tests only assert expect(() => result.current(users)).not.toThrow()

  Nit

  3. Wildcard MSW patterns — ~15 files use '*/endpoint/' instead of edaAPI tagged templates for URL matching

  Path to 10/10

  1. Replace fireEvent with userEvent.setup() + await user.click(), and setTimeout waits with waitFor (+1 pt)
  2. Replace export-only tests with behavioral assertions (+0.5 pt)
  3. Replace setTimeout test waits with waitFor (+0.5 pt)

  What's Good

  Strong MSW adoption throughout (unlike the AWX companion PR), clean factory patterns, good edge case coverage (loading
  states, error states, permission warnings), and well-structured complex tests (SourceEventMapFields.test.tsx,
  useCopyRulebookActivation.test.tsx).

File: frontend/eda/access/roles/hooks/useEdaRoleActions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+  it('should disable Edit for managed roles', () => {
+    const onComplete = vi.fn();
+    const { result } = renderHook(() => useEdaRoleRowActions(onComplete), { wrapper });
+
+    const editAction = result.current.find((a) => 'label' in a && a.label === 'Edit role');
+
+    const managedRole = { id: 1, managed: true } as EdaRbacRole;
+
+    if (editAction && 'isDisabled' in editAction && typeof editAction.isDisabled === 'function') {
```
Comment (prat98) [inline frontend/eda/access/roles/hooks/useEdaRoleActions.test.tsx]: **[Major — Functionality]** Conditional assertion silently passes when the `if` condition is false. This test claims to verify "should disable Edit for managed roles" but if the action shape doesn't match, the assertion is silently skipped and the test passes without checking anything.

Same pattern at line 106 for the delete action.

**Fix:** Assert unconditionally:
```suggestion
    expect(editAction).toBeDefined();
    expect(typeof (editAction as { isDisabled: unknown }).isDisabled).toBe('function');
    expect((editAction!.isDisabled as (item: EdaRbacRole) => string)(managedRole)).toBe(
```

File: frontend/eda/access/credentials/CredentialFormTypes.test.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  });
+
+  it('should handle undefined credential type', () => {
+    const { container } = render(
+      <FormWrapper>
+        <CredentialFormInputs credentialType={undefined} />
+      </FormWrapper>
+    );
+
+    expect(container.children.length).toBeGreaterThanOrEqual(0);
```
Comment (prat98) [inline frontend/eda/access/credentials/CredentialFormTypes.test.tsx]: **[Major — Functionality]** Trivially passing assertion — `container.children.length >= 0` is always true for any rendered output. This test "should handle undefined credential type" cannot fail.

```suggestion
    expect(container.innerHTML).toBe('')
```

Or assert the specific empty-state behavior if the component renders something.

File: frontend/eda/rulebook-activations/RulebookActivationPage/RuleBookActivationTeamAccess.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <Routes>
+          <Route
+            path="/rulebook-activations/:id/team-access"
+            element={<RulebookActivationTeamAccess />}
+          />
+        </Routes>
+      </MemoryRouter>
+    );
+
+    expect(container).toBeTruthy();
```
Comment (prat98) [inline frontend/eda/rulebook-activations/RulebookActivationPage/RuleBookActivationTeamAccess.test.tsx]: **[Major — Functionality]** `expect(container).toBeTruthy()` always passes since `render()` always returns a truthy container. This test "should render without errors" provides no meaningful assertion.

**Fix:** Assert on actual rendered content, e.g.:
```typescript
expect(screen.getByText('Assign teams')).toBeInTheDocument();
```

File: frontend/eda/access/credential-types/hooks/useDeleteCredentialTypes.test.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+});
+
+const server = setupServer();
+
+describe('useDeleteCredentialTypes', () => {
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  const onComplete = vi.fn();
```
Comment (prat98) [inline frontend/eda/access/credential-types/hooks/useDeleteCredentialTypes.test.tsx:40]: **[Major — Functionality]** `onComplete` is a `vi.fn()` declared at `describe` scope but never cleared between tests. Later tests that check `expect(onComplete).toHaveBeenCalled()` may carry stale state from earlier tests.

Same pattern in `useDeleteEdaRoles.test.tsx`.

**Fix:** Add `afterEach(() => { onComplete.mockClear(); })` or declare `onComplete` inside each `it` block.

File: frontend/eda/access/organizations/hooks/useDeleteOrganizations.test.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      created: '2024-01-01T00:00:00Z',
+      modified: '2024-01-01T00:00:00Z',
+      ...overrides,
+    }) as EdaOrganization;
+
+  it('should return a function', () => {
+    const onComplete = vi.fn();
+    const { result } = renderHook(() => useDeleteOrganizations(onComplete), { wrapper });
+
+    expect(typeof result.current).toBe('function');
```
Comment (prat98) [inline frontend/eda/access/organizations/hooks/useDeleteOrganizations.test.tsx:57]: **[Major — Quality]** Low-value test — verifying that a hook "returns a function" adds coverage numbers but no regression protection. This pattern repeats across `useDeleteTeams`, `useDeleteProjects`, `useDeleteCredentials`, `useCopyCredential`, and `useCredentialsTestModal`.

This PR already has great behavioral test examples to follow: `useDeleteCredentialTypes.test.tsx` tests dialog rendering and confirmation flow; `useCopyRulebookActivation.test.tsx` tests success/failure alerts and callbacks.

**Fix:** Add behavioral assertions — verify the dialog opens with correct items, confirm callback fires on user action, verify API calls are made.

File: frontend/eda/access/roles/hooks/EdaContentType.test.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable i18next/no-literal-string */
+import { describe, expect, it } from 'vitest';
+import { EdaContentType } from './EdaContentType';
+
+describe('EdaContentType', () => {
+  it('should have Activation content type', () => {
+    expect(EdaContentType.Activation).toBe('eda.activation');
```
Comment (prat98) [inline frontend/eda/access/roles/hooks/EdaContentType.test.ts:24]: **[Major — Quality]** Testing static constant values — if someone changes `EdaContentType.Activation`, both this test and the constant must update together. The test adds no safety net.

**Fix:** Remove this file, or replace with a single structural test:
```typescript
it('should export all expected content types', () => {
  expect(Object.keys(EdaContentType)).toEqual(expect.arrayContaining([
    'Activation', 'AuditRule', 'Credential', ...
  ]));
});
```

**Review by prat98 — CHANGES_REQUESTED** [review]
AI-assisted code review — 6 inline comments posted. See summary comment for the full review with scores and path to 10/10.

**Comment by prat98** [comment]
## Code Review: test(eda): add vitest coverage for access/, rulebook-activations/ & projects/ (AAP-70845)

### Verdict: NEEDS_CHANGES

### Scores
| Lens | Score | Findings |
|---|---|---|
| Functionality | 3.5/10 | 4 Major, 1 Minor |
| Security | 10/10 | None |
| Quality | 7/10 | 2 Major, 2 Minor, 2 Nit |
| **Overall** | **6.8/10** | |

### Key Findings

#### Major

1. **[Functionality] useEdaRoleActions.test.tsx:90,106** — Conditional assertions wrapped in `if` guards silently skip when condition is false. Tests claim to check "disable Edit/Delete for managed roles" but may never actually assert.

2. **[Functionality] CredentialFormTypes.test.tsx:155** — `expect(container.children.length).toBeGreaterThanOrEqual(0)` is always true — test cannot fail.

3. **[Functionality] RuleBookActivationTeamAccess.test.tsx:55** — `expect(container).toBeTruthy()` always passes since `render()` always returns a container.

4. **[Functionality] useDeleteCredentialTypes.test.tsx:45, useDeleteEdaRoles.test.tsx** — Shared `vi.fn()` mock at describe scope never cleared between tests.

5. **[Quality] useDeleteOrganizations, useDeleteTeams, useDeleteProjects, useCopyCredential, useCredentialsTestModal** — Low-value tests that only verify hooks "return a function" or "don't throw". Good behavioral test patterns already exist in this PR (`useCopyRulebookActivation`, `useDeleteCredentialTypes`) but aren't followed consistently.

6. **[Quality] EdaContentType.test.ts** — Tests that static enum values equal hardcoded strings — adds no safety net.

#### Minor
- Dead MSW handler with double-slash URL in ActivationInstancePage.test.tsx
- Duplicated render blocks in CredentialPlugins.test.tsx
- Inconsistent `onUnhandledRequest` strategy (`'warn'` vs `'bypass'`)

### Positive Observations
- Consistent `userEvent.setup()` — no `fireEvent` anywhere
- Proper MSW lifecycle in every file
- Several files are excellent: `useCopyRulebookActivation`, `SourceEventMapFields`, `ActivationsToolbar`
- Good edge case coverage (null IDs, empty lists, API errors)

### Path to 10/10
Fix the 4 trivially-passing / silently-skipped assertions (+5 pts), upgrade low-value hook tests to match the quality bar set by the stronger files (+1 pt), replace constant-value tests with structural check (+1 pt).

---

*AI-assisted review — see inline comments for details. LOW-confidence findings and items noted as "Needs Human Judgment" require human verification.*

Assisted-by: Claude Code / Opus 4.6 (Anthropic)

File: frontend/eda/access/roles/hooks/useEdaRoleActions.test.tsx (hunk 1 above)
Comment (acruzgon) [reply frontend/eda/access/roles/hooks/useEdaRoleActions.test.tsx]: Fixed in d14230e8d — removed `if` guards, now asserts unconditionally with `expect(editAction).toBeDefined()` followed by direct type-cast assertions.

File: frontend/eda/access/credentials/CredentialFormTypes.test.tsx (hunk 2 above)
Comment (acruzgon) [reply frontend/eda/access/credentials/CredentialFormTypes.test.tsx]: Fixed in d14230e8d — changed to `expect(container.innerHTML).toBe('')` which actually verifies the component renders nothing when credentialType is undefined.

File: frontend/eda/rulebook-activations/RulebookActivationPage/RuleBookActivationTeamAccess.test.tsx (hunk 3 above)
Comment (acruzgon) [reply frontend/eda/rulebook-activations/RulebookActivationPage/RuleBookActivationTeamAccess.test.tsx]: Fixed in d14230e8d — replaced with `waitFor` + `screen.getByText()` assertion on actual rendered content.

File: frontend/eda/access/credential-types/hooks/useDeleteCredentialTypes.test.tsx (hunk 4 above)
Comment (acruzgon) [reply frontend/eda/access/credential-types/hooks/useDeleteCredentialTypes.test.tsx:40]: Fixed in d14230e8d — added `onComplete.mockClear()` in `afterEach` for both useDeleteCredentialTypes and useDeleteEdaRoles.

File: frontend/eda/access/organizations/hooks/useDeleteOrganizations.test.tsx (hunk 5 above)
Comment (acruzgon) [reply frontend/eda/access/organizations/hooks/useDeleteOrganizations.test.tsx:57]: Fixed in d14230e8d — rewrote all 6 low-value tests with PageDialogProvider wrapper and meaningful dialog-content assertions (title text, action buttons, item names). Follows the pattern from useDeleteCredentialTypes.test.tsx.

File: frontend/eda/access/roles/hooks/EdaContentType.test.ts (hunk 6 above)
Comment (acruzgon) [reply frontend/eda/access/roles/hooks/EdaContentType.test.ts:24]: Fixed in d14230e8d — consolidated individual value tests into a structural `expect.arrayContaining` check plus grouped value assertions.

**Review by prat98 — APPROVED** [review]
